lib/puppet/functions/vault_lookup/lookup.rb

```diff
@@ -19,7 +20,9 @@ def lookup(path, vault_url = nil)
     raise Puppet::Error, "Unable to parse a hostname from #{vault_url}" unless uri.hostname
 
     use_ssl = uri.scheme == 'https'
-    connection = Puppet::Network::HttpPool.http_instance(uri.host, uri.port, use_ssl)
+    ssl_context = create_ssl_context(verify_ssl)
```

Comment:
I think it should be nil when there's no ssl
```suggestion
    ssl_context = use_ssl ? create_ssl_context(verify_ssl) : nil
```